### PR TITLE
repository check

### PR DESCRIPTION
**Pull request by saumanbiswas**

### CODE REVIEW

File: pkg/cmd/git/diff.go [hunk 1]
```diff
@@ -80,3 +82,42 @@ func autoDetect() string {
...
+func ensureRepo() {
+	output, err := execLocal("git ls-remote --get-url origin")
+	if err != nil {
+		term.Fatalln("Unsupported Git Hosting provider.")
+	}
+	auth := cli.GetAuthOrDie()
+	if strings.HasSuffix(getHost(output), auth.Network.TeamUrls.BaseAddr) {
+		return
+	}
+	if strings.HasSuffix(getHost(output), strings.TrimPrefix(getHost(auth.ApiServer), "api.")) {
```
Comment (TamalSaha) [inline pkg/cmd/git/diff.go]: Use auth.Network.PublicUrls.BaseAddr

File: pkg/cmd/git/diff.go [hunk 2]
```diff
@@ -80,3 +82,42 @@ func autoDetect() string {
...
+		return
+	}
+	term.Fatalln(fmt.Sprintf("Unsupported Git Hosting provider, or Check your login"))
+}
+
+func getHost(s string) string {
```
Comment (TamalSaha) [inline pkg/cmd/git/diff.go]: Can you add some tests?

File: pkg/cmd/git/git_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+type ensureRepoCase struct {
+	gitRemoteURL string
+	expected     string
+}
+
+func TestGetHost(t *testing.T) {
+	gitURLs := []ensureRepoCase{
+		{
+			gitRemoteURL: "https://appscode.appscode.io/diffusion/43/appscode.git",
```
Comment (sadlil) [inline pkg/cmd/git/git_test.go]: Can we use fake urls fro test? not the real URLs.
ie:
```
https://fake.example.org/something/more/something
```

File: pkg/cmd/git/diff.go [hunk 4]
```diff
@@ -80,3 +82,42 @@ func autoDetect() string {
...
 	return ""
 }
+
+func ensureRepo() {
+	output, err := execLocal("git ls-remote --get-url origin")
+	if err != nil {
+		term.Fatalln("Unsupported Git Hosting provider.")
+	}
+	auth := cli.GetAuthOrDie()
+	if strings.HasSuffix(getHost(output), auth.Network.TeamUrls.BaseAddr) {
```
Comment (TamalSaha) [inline pkg/cmd/git/diff.go]: It should be 

`strings.HasSuffix(getHost(output), "." + auth.Network.TeamUrls.BaseAddr)`

File: pkg/cmd/git/diff.go [hunk 5]
```diff
@@ -80,3 +82,42 @@ func autoDetect() string {
...
+		return
+	}
+	if strings.HasSuffix(getHost(output), auth.Network.PublicUrls.BaseAddr) {
```
Comment (TamalSaha) [inline pkg/cmd/git/diff.go]: It should be 
`strings.HasSuffix(getHost(output), "." + auth.Network.PublicUrls.BaseAddr)`

**Pull request merged**